File: Integration Examples/eAPI v1.8/java/eAPI-v18-connector/src/main/java/cz/monetplus/mips/eapi/v18/connector/entity/ext/MaskedClnRPExtensionV16.java
package cz.monetplus.mips.eapi.v18.connector.entity.ext;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class MaskedClnRPExtensionV16 extends Extension {

	private static final long serialVersionUID = -3825192932302805075L;
	
	@XmlElement
	public String extension = PrivateApiExtension.RECURRENT_PAYMENT_MASK_CLN.getCode();

	@XmlElement
	public String maskedCln;

	@XmlElement
	public String longMaskedCln;

	@XmlElement
	public String expiration;

	public MaskedClnRPExtensionV16() {
	}

	public MaskedClnRPExtensionV16(String dttm, String maskedCln, String longMaskedCln, String expiration) {
		this.dttm = dttm;
		this.maskedCln = maskedCln;
		this.longMaskedCln = longMaskedCln;
		this.expiration = expiration;
	}

	@Override
	public String toSign() {
		StringBuilder sb = new StringBuilder(); 
		add(sb, extension);
		add(sb, dttm);
		add(sb, maskedCln);
		add(sb, expiration);
		add(sb, longMaskedCln);
		deleteLast(sb);
		return sb.toString();
	}

}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/Ext/MaskedClnRPExtensionV16.cs
using System.Text;
using System.Xml.Serialization;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects.Ext;

[XmlRoot("maskClnRP")]
public class MaskedClnRPExtensionV16 : Extension
{
    [XmlElement(ElementName = "extension")] [JsonProperty("extension")]
    public string Extension = PrivateApiExtension.RECURRENT_PAYMENT_MASK_CLN;

    public MaskedClnRPExtensionV16()
    {
    }

    public MaskedClnRPExtensionV16(string dttm, string? maskedCln, string? longMaskedCln, string? expiration)
    {
        Dttm = Dttm;
        MaskedCln = maskedCln;
        LongMaskedCln = longMaskedCln;
        Expiration = expiration;
    }

    //@XmlElement
    [XmlElement(ElementName = "maskedCln")]
    [JsonProperty("maskedCln")]
    public string? MaskedCln { get; set; }

    //@XmlElement
    [XmlElement(ElementName = "longMaskedCln")]
    [JsonProperty("longMaskedCln")]
    public string? LongMaskedCln { get; set; }

    //@XmlElement
    [XmlElement(ElementName = "expiration")]
    [JsonProperty("expiration")]
    public string? Expiration { get; set; }

    //@Override
    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, extension);
        Add(sb, Dttm);
        Add(sb, MaskedCln);
        Add(sb, Expiration);
        Add(sb, LongMaskedCln);
        DeleteLast(sb);
        return sb.ToString();
    }
}